sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/IntegrationTestBase.java

```diff
@@ -128,17 +137,55 @@ static String getConnectionString() {
     }
 
     static String getConnectionString(boolean withSas) {
+        String connectionString = GLOBAL_CONFIGURATION.get(EVENT_HUB_CONNECTION_STRING_ENV_NAME);
         if (withSas) {
-            return System.getenv(EVENT_HUB_CONNECTION_STRING_WITH_SAS);
+            final String shareAccessSignatureFormat = "SharedAccessSignature sr=%s&sig=%s&se=%s&skn=%s";
```

Comment:
Is there a reason we are calculating the SAS here? The intention was that if the test environment variable had a SAS key in it, we'd run that version of the test.

Reply:
The test case  `EventHubAsyncClientIntegrationTest.sendAndReceiveEventByAzureSasCredential` and `EventHubProducerAsyncClientIntegrationTest.sendWithSasConnectionString` need to use Sas connection, but the relevant parameter is not recorded in `test-resources.json`, so added the calculation of `SharedAccessSignature`.

Reply:
I meant that we don't want to duplicate this logic. I typed out what I was hinting at.... we don't want to copy this entire logic and use existing classes to get the shared access signature.

```java
var connectionProperties = new ConnectionStringProperties(connectionString);
var credential = new EventHubSharedKeyCredential(connectionProperties.getSharedAccessKeyName(), connectionProperties.getSharedAccessKey());

var tokenManager = new AzureTokenManagerProvider(
    CbsAuthorizationType.SHARED_ACCESS_SIGNATURE, connectionProperties.getFullyQualifiedNamespace(), ClientConstants.AZURE_ACTIVE_DIRECTORY_SCOPE);

var scopes = tokenManager.getScopesFromResource(resourceUrl);
var tokenRequest = new TokenRequestContext().addScopes(scopes);

// This contains the shared access signature.
Mono<AccessToken> accessTokenMono = credential.getToken(tokenRequest);
AccessToken accessToken = accessTokenMono.block(Duration.ofSeconds(5));

assertNotNull(accessToken);

String sharedAccessSignature = accessToken.getToken();
```

Reply:
According to your comment, fixed in the new version.